batchflow/models/eager_torch/base.py

```diff
@@ -0,0 +1,1203 @@
+""" Eager version of TorchModel. """
+import os
+import re
+import threading
+import inspect
+from collections import OrderedDict
+from functools import partial
+from pprint import pprint
+
+import numpy as np
+import torch
+import torch.nn as nn
+from torch.utils.tensorboard import SummaryWriter
+
+from .utils import unpack_fn_from_config, get_shape
+from .layers import ConvBlock
+from .losses import CrossEntropyLoss
+from ... import Config
+
+
+
+LOSSES = {
+    'mse': nn.MSELoss,
+    'bce': nn.BCEWithLogitsLoss,
+    'ce': CrossEntropyLoss,
+    'crossentropy': CrossEntropyLoss,
+    'absolutedifference': nn.L1Loss,
+    'l1': nn.L1Loss,
+    'cosine': nn.CosineSimilarity,
+    'cos': nn.CosineSimilarity,
+    'hinge': nn.HingeEmbeddingLoss,
+    'huber': nn.SmoothL1Loss,
+    'logloss': CrossEntropyLoss,
+}
+
+DECAYS = {
+    'exp': torch.optim.lr_scheduler.ExponentialLR,
+    'lambda': torch.optim.lr_scheduler.LambdaLR,
+    'step': torch.optim.lr_scheduler.StepLR,
+    'multistep': torch.optim.lr_scheduler.MultiStepLR,
+    'cos': torch.optim.lr_scheduler.CosineAnnealingLR,
+}
+
+DECAYS_DEFAULTS = {
+    torch.optim.lr_scheduler.ExponentialLR : dict(gamma=0.96),
+    torch.optim.lr_scheduler.LambdaLR : dict(lr_lambda=lambda epoch: 0.96**epoch),
+    torch.optim.lr_scheduler.StepLR: dict(step_size=30),
+    torch.optim.lr_scheduler.MultiStepLR: dict(milestones=[30, 80]),
+    torch.optim.lr_scheduler.CosineAnnealingLR: dict(T_max=None)
+}
+
+
+
+class EagerTorch:
+    r""" Base class for eager Torch models.
+
+    Parameters
+    ----------
+    inputs : dict, optional
+        Mapping from placeholder names (e.g. ``images``, ``labels``, ``masks``) to arguments of their initialization.
+        Allows to create placeholders of needed shape and data format and initialize model before
+        first pass of actual batch data (thus explicitly imposing shapes).
+
+        Value must be a dictionary with parameters. If some parameters are omitted, then defaults will be at use.
+
+            dtype : str or torch.dtype
+                Data type. Default is 'float32'.
+
+            shape : int, None, sequence of ints or Nones
+                Tensor shape with channels and without batch size. Default is None.
+
+            classes : int, array-like or None
```

Comment:
This is for classification, what about segmentation?

Also, in current configs number of classes is defined in `'inputs/labels/classes'` section, and in this docstring it is in `'inputs/classes'` - did you change this,. or it's just an error?

All in all, all this section needs clarification